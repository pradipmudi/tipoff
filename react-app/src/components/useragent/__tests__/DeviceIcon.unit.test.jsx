import React from "react";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TestDeviceIcon from '../DeviceIcon';

describe("DeviceIcon", () => {

    it("matches snapshot", () => {
        const device = {
            type : 'mobile',
            name : 'S8',
            vendor : 'samsung'
        }
        const tree = renderer.create(<TestDeviceIcon device={device} />).toJSON();
        expect(tree).toMatchSnapshot();
    
    });
    
    it("renders without issues", () =>{
        const device = {
            type : 'mobile',
            name : 'S8',
            vendor : 'samsung'
        }
        const component = shallow(<TestDeviceIcon device={device} />);
        expect(component.find('.fas').hasClass('fa-mobile')).toEqual(true);
        expect(component.find('.fas').props().title)
            .toEqual(`${device.type}:${device.name}:${device.vendor}`);
    });

    it("renders with default icon when browser is not passed", () => {
        const component = shallow(<TestDeviceIcon  />);
        expect(component.find('.fas').hasClass('fa-laptop')).toEqual(true);
    })

});
