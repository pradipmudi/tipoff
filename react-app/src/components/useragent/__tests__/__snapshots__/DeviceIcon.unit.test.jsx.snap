// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceIcon matches snapshot 1`] = `
<span>
  <span
    className="icon"
    title="mobile:S8:samsung"
  >
    <i
      aria-hidden="true"
      className="fas fa-mobile"
    />
  </span>
</span>
`;
